### PR TITLE
Fix LTC Decoupling

### PR DESCRIPTION
**Pull request by antoniojkim**
We previously had to merge in some temporary "hacks" to decouple LTC from the TS backend. Now that the proper fix has been merged into PyTorch master, we need to fix up this branch to use those changes.

Also, added PyTorch submodule for codegen purposes. Previously we were assuming that a pytorch clone was located in a specific place relative to `torch_mlir`. However, that is not an assumption that we could generally make. Adding PyTorch as a submodule fixes this.

CC: @ke1337 @wconstab

### CODE REVIEW

File: python/torch_mlir/csrc/base_lazy_backend/mlir_node.h [hunk 1]
```diff
@@ -27,10 +27,63 @@ namespace lazy {
...
+
+  hash_t hash() const override;
+
+  hash_t shapeHash() const override;
 
   virtual TorchMlirOpVector
   Lower(TorchMlirFunction function, TorchMlirLoweringContext* loctx) const;
+
+private:
+  // The hash of the dag WITH size info. Used for shape caching
```
Comment (henrytwo) [inline python/torch_mlir/csrc/base_lazy_backend/mlir_node.h:53]: nit: missing period at end of line

**Review by henrytwo — APPROVED** [review]

**Comment by antoniojkim** [comment]
@silvasean Gentle reminder to please review this when you can

File: python/torch_mlir/csrc/base_lazy_backend/dynamic_ir.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+// This file is adapted from pytorch/pytorch
+// https://github.com/pytorch/pytorch/blob/master/torch/csrc/lazy/ts_backend/dynamic_ir.cpp
+//===----------------------------------------------------------------------===//
+
+#include <torch/csrc/lazy/ts_backend/dynamic_ir.h>
+
+namespace torch {
+namespace lazy {
+
+DimensionNode::DimensionNode(OpKind op, OpList operands, hash_t hash_seed)
```
Comment (silvasean) [inline python/torch_mlir/csrc/base_lazy_backend/dynamic_ir.cpp:18]: Is there any doc on the strategy here for dynamic shapes?

**Review by silvasean — APPROVED** [review]

File: python/torch_mlir/csrc/base_lazy_backend/dynamic_ir.cpp (hunk 2 above)
Comment (antoniojkim) [reply python/torch_mlir/csrc/base_lazy_backend/dynamic_ir.cpp:18]: There is, but its very much a work in progress I believe. This class is just a placeholder introduced for now. We'll need to make further changes to this once they finalize the design and implemention in PyTorch master.

**Pull request merged**